go-controller/pkg/ovn/gateway.go

```diff
@@ -386,6 +386,11 @@ func (gw *GatewayManager) GatewayInit(
 			types.NetworkExternalID:  gw.netInfo.GetNetworkName(),
 			types.TopologyExternalID: gw.netInfo.TopologyType(),
 		}
+		if gw.netInfo.IsPrimaryNetwork() && gw.netInfo.TopologyType() == types.Layer2Topology {
```

Comment:
@kyrtapz / @maiqueb / @qinqon : Should we do this only when at least 1 VM pod is present in this L2 UDN network? Meaning do an "update to LRP" as day2... I am not sure if we should always be enabling this for all users cause we only set DHCP options for LSPs of VM pods, so unsure if sending periodic RAs for non-VM pods is needed? Or is there an existing virt feature gate we can leverage?

Reply:
I don't see much use of L2 for pods, not sure if it's worth it, I mean it will affect pods within layer2, I think this can be fixed later on, if users has issue with this at pods. It will mean that there are l2+pods users.

Reply:
I think long-term, what would make sense it to expose this functionality using yet another knob in the network configuration ...

We should probably refactor this.